errors.go

```diff
@@ -335,3 +335,29 @@ func Find(origErr error, test func(error) bool) error {
 	})
 	return foundErr
 }
+
+// Chain ties together two errors, giving them a Causer interface and a concatenated Error message.
+type chain struct {
+	currentErr error
+	prevErr    error
+}
+
+func (err *chain) Error() string {
+	return err.currentErr.Error() + "\n" + err.prevErr.Error()
```

Comment:
Error should not got preErr.

e.g. TruncateErr, error info should only contain `a column, b row value be truncate` for end-user

Reply:
Okay, that makes sense since this is joining two separate errors. The previous will show up in the formatter.